### PR TITLE
Qt: update developer page when it's visible

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Can we only update this when settings is opened? We're already using a bunch of timers, and I think the number of timers can easily get out of control since they're so easy to use.

File: selfdrive/ui/qt/offroad/settings.cc [hunk 1]
```diff
@@ -217,37 +222,55 @@ QWidget * device_panel() {
...
+}
+
+void DeveloperPanel::updatePanel() {
+  auto dev_params = getParams();
+  for (auto &[name, value] : dev_params) {
+    labels.at(QString::fromStdString(name))->setContent(QString::fromStdString(value));
+  }
+}
+
+std::vector<std::pair<std::string, std::string>> DeveloperPanel::getParams() {
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/settings.cc:260]: why not put this in updatePanel?

File: selfdrive/ui/qt/offroad/settings.cc [hunk 2]
```diff
@@ -217,37 +222,55 @@ QWidget * device_panel() {
...
     }
   )");
-  return widget;
+}
+
+void DeveloperPanel::showEvent(QShowEvent *event) {
+  updatePanel();
+}
+
+void DeveloperPanel::updatePanel() {
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/settings.cc:253]: rename to update? we'll need update functions for a lot of our widgets, so think it makes sense to call them all update

File: selfdrive/ui/qt/offroad/settings.cc (hunk 1 above)
Comment (deanlee) [reply selfdrive/ui/qt/offroad/settings.cc:260]: Because I also create widgets from params in the constructor to simplify the initialization.

**Pull request merged**